### PR TITLE
allow service account create projects

### LINKED ISSUE #6984: Add new group for Service Account in the API

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: allow service account to create projects when belongs to projectmanagers group

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6984

```release-note
allow service account to create projects when belongs to projectmanagers group
```

### CODE REVIEW

File: pkg/handler/v1/project/project.go [hunk 1]
```diff
@@ -85,6 +88,58 @@ func CreateEndpoint(projectProvider provider.ProjectProvider, privilegedProjectP
...
+
+	if len(bindings) == 0 {
+		return nil, errors.New(http.StatusInternalServerError, fmt.Sprintf("no bindings for service account user %s", saEmail))
+	}
+	saBinding := bindings[0]
+	if !strings.HasPrefix(saBinding.Spec.Group, rbac.ProjectManagerGroupNamePrefix) {
+		return nil, errors.New(http.StatusForbidden, "the Service Account is not allowed to create a project")
+	}
+	if len(projectReq.Body.Users) == 0 {
+		return nil, errors.New(http.StatusBadRequest, "expected user emails")
```
Comment (lsviben) [inline pkg/handler/v1/project/project.go]: you could change the message here to something like:

"owner user email list required for project creation through Service Account"

Comment (zreigz) [reply pkg/handler/v1/project/project.go]: done

**Pull request merged**